### PR TITLE
Fix wrong Region Info Names

### PR DESCRIPTION
**Pull request by tarekgh**
We had some typo in some of the region info data. The change here is fixing that.

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System.Globalization.Tests/System/Globalization/RegionInfoTests.cs [hunk 1]
```diff
@@ -266,5 +267,19 @@ public void ValidateThrowingWhenUsingCustomUnspecifiedLcid()
 
             AssertExtensions.Throws<ArgumentException>("culture", () => new RegionInfo(4096));
         }
+
+        [Fact]
+        public void BuiltInRegionListTest()
+        {
+            // Ensure we can create all region info objects from the built-in list
+            Dictionary<string, string> regionNames = (Dictionary<string, string>)typeof(RegionInfo).Assembly.GetType("System.Globalization.CultureData").GetProperty("RegionNames", BindingFlags.Static | BindingFlags.NonPublic).GetValue(null);
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System.Globalization.Tests/System/Globalization/RegionInfoTests.cs:275]: Is there another source we could be enumerating? e.g. how does this related to the cultures enumerable from CultureInfo.GetCultures(All)... if we were to parse the regions out of those, would that be equivalent to (some of) this data?

Comment (tarekgh) [reply src/libraries/System.Runtime/tests/System.Globalization.Tests/System/Globalization/RegionInfoTests.cs:275]: Culture enumeration can be done too but in our data, we add some data that is not getting from the enumeration. That is why I decided to do use the data we build.

**Review by ericstj — APPROVED** [review]

**Pull request merged**